### PR TITLE
Added RethinkDB storage option & fixed config to use proper JSON

### PR DESCRIPTION
**Pull request by PassTheMayo**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var connect_rate_limit = require('connect-ratelimit');
 var DocumentHandler = require('./lib/document_handler');
 
 // Load the configuration and set some defaults
-var config = JSON.parse(fs.readFileSync('./config.js', 'utf8'));
+var config = require('./config.json');
```
Comment (seejohnrun) [inline server.js]: This change isn't backward compatible for existing installations.  Also it seems unrelated to the overall change

File: lib/document_stores/rethinkdb.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+var crypto = require('crypto');
+var rethink = require('rethinkdbdash');
+
+var RethinkDBStore = function (options) {
+  this._options = options;
```
Comment (seejohnrun) [inline lib/document_stores/rethinkdb.js]: maybe instead we can write this as:

``` js
this.client = rethink({
  host: options.host || '127.0.0.1'
  ...
})
```

it looks like options isn't used outside of the initializer and this is fairly verbose

File: lib/document_stores/rethinkdb.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+var crypto = require('crypto');
```
Comment (seejohnrun) [inline lib/document_stores/rethinkdb.js]: can use es6 in this file

File: lib/document_stores/rethinkdb.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      callback(true);
+    });
+  } catch (err) {
+    callback(false);
+  }
+};
+
+RethinkDBStore.prototype.get = function (key, callback) {
+  this.client.table('uploads').get(RethinkDBStore.md5(key)).run((error, result) => {
+    if (error || !result) return callback(false);
```
Comment (seejohnrun) [inline lib/document_stores/rethinkdb.js]: I don't know for sure, but `!result` doesn't need to indicate failure does it?

File: lib/document_stores/rethinkdb.js (hunk 3 above)
Comment (PassTheMayo) [reply lib/document_stores/rethinkdb.js]: The rest of the haste server uses require(), are you sure?

File: lib/document_stores/rethinkdb.js (hunk 2 above)
Comment (PassTheMayo) [reply lib/document_stores/rethinkdb.js]: Definetly, I'll get to that once I get time

File: lib/document_stores/rethinkdb.js (hunk 4 above)
Comment (PassTheMayo) [reply lib/document_stores/rethinkdb.js]: `!result` will be true when thay is not a result that has been pasted. I copied everything from the file storage system, and when a file didn't exist, it called `callback(false)`.

File: lib/document_stores/rethinkdb.js (hunk 3 above)
Comment (seejohnrun) [reply lib/document_stores/rethinkdb.js]: Sorry this comment is a bit confusing - `require` is fine but I'm slowly updating things to use ES6 syntax like `const / let`, ` => () `, etc

File: server.js (hunk 1 above)
Comment (PassTheMayo) [reply server.js]: Oh, ok

File: lib/document_stores/rethinkdb.js (hunk 3 above)
Comment (PassTheMayo) [reply lib/document_stores/rethinkdb.js]: Ooohhh, ok. I'll do that when I get the chance.

**Comment by PassTheMayo** [comment]
Just added some more changes, is that good?

File: config.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (seejohnrun) [inline config.json]: This file should be reverted to its previous state

**Comment by PassTheMayo** [comment]
There ya go!

**Comment by PassTheMayo** [comment]
:D
